### PR TITLE
added login/signup page with basic functionality

### PR DESCRIPTION
**Pull request by Raghuboi**
- login/signup page with basic functionality, query the database to register/authenticate a user

- password hashing with bcrypt

- currently there is no validation for email and password in the sign up form, it just inserts the user into the db as long as the password and confirm password fields match

- will have to create sessions in the future, add form validation and send authentication links to emails

### CODE REVIEW

File: web/utils/query.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    password: process.env.DB_PASS,
+    database: process.env.DB_DATABASE,
+    waitForConnections: true,
+    connectionLimit: 10
+})
+
+export async function query(q, values) {
+  try {
+    const results = await pool.query(q, values)
+    await pool.end()
```
Comment (xTrig) [inline web/utils/query.js:16]: We actually don't want to ever close this pool. It should always be available for the lifecycle of the server.

File: web/utils/query.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require('dotenv').config(); //Load from the .env file
```
Comment (xTrig) [inline web/utils/query.js:1]: This is loaded from the new server.js file, we don't need this anymore.

File: web/utils/query.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const mysql = require('mysql2');
```
Comment (xTrig) [inline web/utils/query.js:2]: Maybe import mysql2/promise so we can use async/await with the pools/queries

File: web/pages/api/signin.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    .post(async (req, res) => {
+        const { email, password } = req.body 
+
+        if (!email || !password) res.status(400).json({ error: 'Email or Password cannot be blank' })
+
+        try {
+            const results = await query(`
+              SELECT password FROM users WHERE email = ?
+            `, email)
```
Comment (xTrig) [inline web/pages/api/signin.js:15]: Prepared Statements have to pass in their fields in an array, so you'd have to wrap email in an array.

`[email]`

**Review by xTrig — CHANGES_REQUESTED** [review]

File: web/pages/api/signup.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+          /*
+            todo:
+              - validate password: see if it has 8 digits and 1 symbol 
+              - add check to see if email has @uwinnipeg.ca or @webmail.uwinnipeg.ca
+              - add check to see if email exists
+              - send validation link to email
+          */
+
+            const hashedPW = await bcrypt.hash(password, 10)
```
Comment (xTrig) [inline web/pages/api/signup.js:22]: The salt rounds should probably be a system constant somewhere, in case we decide to change it later for some reason.

**Review by xTrig — APPROVED** [review]

File: web/pages/api/signup.js (hunk 5 above)
Comment (Raghuboi) [reply web/pages/api/signup.js]: added now

**Pull request merged**